cmd/cloudshell_open/cloudrun.go

```diff
@@ -112,14 +113,24 @@ func tryFixServiceName(name string) string {
 	if name == "" {
 		return name
 	}
+
+	name = strings.ToLower(name)
+
 	if !unicode.IsLetter([]rune(name)[0]) {
 		name = fmt.Sprintf("svc-%s", name)
 	}
+
 	if len(name) > 63 {
 		name = name[:63]
 	}
+
 	for name[len(name)-1] == '-' {
 		name = name[:len(name)-1]
 	}
+
+	reg, _ := regexp.Compile("[^a-z0-9-]+")
+
+	name = reg.ReplaceAllString(name, "-")
```

Comment:
Please move this all the way up below ToLower.

For example if name is `šo`, this will result in `-o`, but if you convert this to `-o` early on, we will prefix it with `svc--o`, at least.